python_katas/kata_1/test.py

```diff
@@ -69,78 +111,190 @@ class TestIsUniqueString(unittest.TestCase):
-        pass
+    def test_prime_1(self):
+        number = 7
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_prime_2(self):
+        number = 17
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_prime_3(self):
+        number = 13
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_not_prime_1(self):
+        number = 1
+        self.assertEqual(questions.prime_number(number), False)
+
+    def test_not_prime_2(self):
+        number = -20
+        self.assertEqual(questions.prime_number(number), False)
+
+    def test_not_prime_3(self):
+        number = 2.4
+        self.assertEqual(questions.prime_number(number), False)
 
 
 class TestPalindromeNum(unittest.TestCase):
     """
     1 Katas
     """
+    def test_pal1True(self):
+        num = 1441
+        self.assertTrue(questions.palindrome_num(num), "the num is palindrome but you return False")
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_pal1True(self):
+        num = 11
+        self.assertTrue(questions.palindrome_num(num), "the num is palindrome but you return False")
+
+    def test_pal1False(self):
+        num = 113
+        self.assertFalse(questions.palindrome_num(num),"the num is not palindrome but you return True")
+
+    def test_pal1Zero(self):
+        num = 0
+        self.assertTrue(questions.palindrome_num(num)," 0 is palindrom num but you return False")
 
 
 class TestPairMatch(unittest.TestCase):
     """
     3 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_Pair1(self):
+        men = {'Ben': 34, 'Ronaldo': 37, 'Ancelotti': 62}
+        women = {'Yasmin': 22, 'Inbar': 18, 'Angelina': 52}
+        self.assertEqual(questions.pair_match(men, women), ('Ancelotti', 'Angelina'))
+
+    def test_Pair2(self):
+        men = {'Roi': 65, 'Eran': 82, 'Ido': 20}
+        women = {'Sivan': 70, 'Orly': 18, 'Neta': 65}
+        self.assertEqual(questions.pair_match(men, women), ('Roi', 'Neta'))
 
 
 class TestBadAverage(unittest.TestCase):
     """
     1 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def easy_peasy(self):
+        nums = [10, 20, 30]
+        self.assertEqual(questions.bad_average(nums), 20)
+
+    def fractions(self):
+        nums = [0.5, 2, 9.5]
+        self.assertEqual(questions.bad_average(nums), 4)
+
+    def zeros(self):
+        nums = [0, 3, 0]
+        self.assertEqual(questions.bad_average(nums), 1)
+
+    def fractions_in_answer(self):
+        nums = [1, 1, 1]
+        self.assertEqual(questions.bad_average(nums), 1/3)
+
+    def mistake(self):
+        nums = [10, 20, 30]
+        self.assertEqual(questions.bad_average(nums), 30)
+
 
 
 class TestBestStudent(unittest.TestCase):
     """
     1 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def best_student_original_example(self):
```

Comment:
each function in the test case class must start with `test_`.
In your case: `test_best_student_original_example`